src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt

```diff
@@ -242,8 +242,14 @@ class ImplLookup(
     private fun rawFindImplsAndTraits(ty: Ty): List<TraitImplSource> {
         val implsAndTraits = mutableListOf<TraitImplSource>()
         when (ty) {
-            is TyTraitObject ->
+            is TyTraitObject -> {
                 ty.trait.flattenHierarchy.mapTo(implsAndTraits) { TraitImplSource.Object(it.element) }
+                findExplicitImpls(ty) {
+                    // skip blanket implementations
+                    if (ty == it.impl.typeReference?.type) implsAndTraits += TraitImplSource.ExplicitImpl(it);
```

Comment:
Why is it needed? I don't understand the comment. Should be already filtered in `findExplicitImpls`. 

Btw `==` should not be used for type comparison in type inference, use `ctx.canCombineTypes()` instead

Reply:
The problem is that when resolving associate items for trait bounds, `selfTy` is also a trait object, and in that case `findExplicitImpls` finds also blanket implementations which are inapplicable to trait bound.
I have tried to fix it, but there is too much code that relies on traits being trait object declarations.

I know that blanket impls can be applicable to actual trait objects, but I also thought that partial implementation is better that nothing, so I have added this condition as a workaround.

Reply:
I fixed it in #4953